ksqldb-functional-tests/src/main/java/io/confluent/ksql/test/tools/TestExecutor.java

```diff
@@ -302,15 +305,23 @@ private static void verifyTopology(final TestCase testCase) {
               + "THIS IS BAD!",
           actualTopology, is(expectedTopology));
 
-      final Map<String, String> generated = testCase.getGeneratedSchemas();
-      for (final Map.Entry<String, String> e : expected.getSchemas().entrySet()) {
+      final Map<String, SchemaNode> generatedSchemas =
+          testCase.getGeneratedSchemas().entrySet().stream()
+              .collect(Collectors.toMap(
+                  Entry::getKey,
+                  e -> SchemaNode.fromPhysicalSchema(e.getValue())));
+      for (final Map.Entry<String, SchemaNode> e : expected.getSchemas().entrySet()) {
```

Comment:
Rather than converting the the PhysicalSchema into a SchemaNode, would it not be better to convert the SchemaNode into the PhysicalSchema and then compare?  I guess it doesn't make much difference, just feels more type safe...

e.g. if someone came along and added a new field to PhysicalSchema, then the comparison would still pass if comparing strings, but would fail if comparing PhysicalSchemas.

`SchemaNode` is basically only there to serialize the parts of `PhysicalSchema`.  So, I'd suggest writing a Jackson serializer and deserializer for `PhysicalSchema`, and dropping `SchemaNode`.  Worth thinking about anyway.  Take a look at `LogicalSchemaSerializer` and `LogicalSchemaDeserializer` for examples of how this is done if you've not done it before.

Reply:
Sounds good, I'll drop SchemaNode in favor of PhysicalSchema. I didn't do that initially since I didn't want to add Jackson annotations to PhysicalSchema, but your suggestion of introducing separate serializers and deserializers works around that. Thanks!